src/include/function/copy/copy_function.h

```diff
@@ -0,0 +1,104 @@
+#pragma once
+
+#include "common/types/value/value.h"
+#include "function/function.h"
+
+namespace kuzu {
+namespace function {
+
+struct CopyFuncLocalState {
+    virtual ~CopyFuncLocalState() = default;
+
+    template<class TARGET>
+    TARGET& cast() {
+        return common::ku_dynamic_cast<CopyFuncLocalState&, TARGET&>(*this);
+    }
+};
+
+struct CopyFuncSharedState {
+    virtual ~CopyFuncSharedState() = default;
+
+    template<class TARGET>
+    TARGET& cast() {
+        return common::ku_dynamic_cast<CopyFuncSharedState&, TARGET&>(*this);
+    }
+};
+
+struct CopyFuncBindData {
+    std::vector<std::string> names;
+    std::vector<common::LogicalType> types;
+    std::string fileName;
+    bool canParallel;
+
+    CopyFuncBindData(std::vector<std::string> names, std::string fileName, bool canParallel)
+        : names{std::move(names)}, fileName{std::move(fileName)}, canParallel{canParallel} {}
+
+    virtual ~CopyFuncBindData() = default;
+
+    void bindDataType(std::vector<common::LogicalType> types_) { types = std::move(types_); }
```

Comment:
looks like this is a simple setter. Why we call it bindXXX? Also, can u do it as part of the constructor. Do we cases where types are not resolved together with names?